packages/webapp/src/members/components/withdraw-funds/withdraw-modal-step-form.tsx

```diff
@@ -0,0 +1,150 @@
+
+    const onChangeFields = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setFields(e);
+    };
+
+    const validateAccountField = (e: React.FormEvent<HTMLInputElement>) => {
+        const target = e.target as HTMLInputElement;
+        if (target.validity.valueMissing) {
+            target.setCustomValidity("Enter an account name");
+        } else {
+            target.setCustomValidity("Invalid account name");
+        }
+    };
+
+    const validateAmountField = (e: React.FormEvent<HTMLInputElement>) => {
+        const target = e.target as HTMLInputElement;
+        if (target.validity.rangeOverflow) {
+            target.setCustomValidity("Insufficient funds available");
+        } else {
+            target.setCustomValidity("Enter a valid withdrawal amount");
+        }
+    };
+
+    const clearErrorMessages = (e: React.FormEvent<HTMLInputElement>) => {
+        (e.target as HTMLInputElement).setCustomValidity("");
+    };
+
+    if (!availableFunds) return null;
+
+    const maxWithdrawal = Number(
+        assetToString(availableFunds!, availableFunds.precision).split(" ")[0]
+    );
+
+    const setMaxAmount = () =>
+        setFields({ target: { id: "amount", value: maxWithdrawal } });
+
+    const isThirdPartyWithdrawal = ualAccount.accountName !== formState[0].to;
+
+    const amountInputPreventChangeOnScroll = (
+        e: React.WheelEvent<HTMLInputElement>
+    ) => (e.target as HTMLInputElement).blur();
+
+    return (
+        <div className="space-y-4">
+            <Heading>Withdraw funds</Heading>
+            <Text>
+                Withdraw my funds from the Eden contract to a public{" "}
+                {tokenConfig.symbol} account.
+            </Text>
+            <Text>
+                Available:{" "}
+                <span className="font-medium">
+                    {assetToLocaleString(
+                        availableFunds,
+                        availableFunds.precision
+                    )}
+                </span>
+            </Text>
+            <form onSubmit={onPreview} className="space-y-3">
+                <Form.LabeledSet
+                    label={`${tokenConfig.symbol} account name (12 characters)`}
+                    htmlFor="to"
+                >
+                    <Form.Input
+                        id="to"
+                        type="text"
+                        required
+                        value={fields.to}
+                        onChange={onChangeFields}
+                        maxLength={12}
+                        pattern="^[a-z,1-5.]{1,12}$"
+                        onInvalid={validateAccountField}
+                        onInput={clearErrorMessages}
+                    />
+                </Form.LabeledSet>
+                <Form.LabeledSet label="Amount to withdraw" htmlFor="amount">
+                    <div className="flex space-x-2">
+                        <div className="relative flex-1">
+                            <Form.Input
```

Comment:
praise: great component to handle assets! 
suggestion: refactor this component to a `Form.Asset` component! I can totally see it being reused in new features!

Reply:
Done ✅